src/wallet/wallet.cpp

```diff
@@ -127,7 +127,10 @@ SaplingPaymentAddress CWallet::GenerateNewSaplingZKey()
     libzcash::SaplingExtendedSpendingKey xsk;
     do
     {
+        // always derive hardened keys
         xsk = m_32h_cth.Derive(hdChain.saplingAccountCounter | ZIP32_HARDENED_KEY_LIMIT);
+        metadata.hdKeypath     = "m/0'/0'/"+std::to_string(hdChain.saplingAccountCounter)+"'";
```

Comment:
This needs to be modified to:
```
metadata.hdKeypath = "m/32'/" + std::to_string(Params().BIP44CoinType()) + "'/" + std::to_string(hdChain.saplingAccountCounter) + "'";
```

You could make `Params().BIP44CoinType()` a variable after `auto m` so you can reuse it.